tuna/example/example_lib.py

```diff
@@ -166,26 +196,81 @@ def get_kwargs(self,
       @param gpu_idx Unique ID of the GPU
       @param f_vals Dict containing process specific runtime information
     """
-    kwargs: Dict[str, Any] = super().get_kwargs(gpu_idx, f_vals)
+    kwargs: Dict[str, Any] = super().get_kwargs(gpu_idx, f_vals, tuning)
 
     return kwargs
 
-  def get_jobs(self,
-               session: DbSession,
-               find_state: List[str],
-               set_state: str,
-               session_id: int,
-               claim_num: int = None,
-               no_update: bool = False):
-    """Get jobs based on find_state"""
-    self.logger.info('Placeholder')
+  def get_job_list(self, session, find_state=None, claim_num=None):
+    """!Get list of jobs
+    @param session DB session
+    @param find_state state of DB job
+    @param claim_num Number of jobs to pick up
+    """
+    job_list = gen_select_objs(session, self.get_job_attr(),
+                               self.dbt.job_table.__tablename__,
+                               "WHERE state='new'")
+    return job_list
 
-    return True
+  def serialize_jobs(self, session, batch_jobs):
+    """!Return list of serialize jobs
+    @param session DB session
+    @param batch_jobs Number of DB jobs
+    """
+    return [elem.to_dict() for elem in batch_jobs]
 
-  def get_context_list(self, session, batch_jobs):
-    """Get a list of context items to be used for celery task"""
-    raise NotImplementedError("Not implemented in example_lib")
+  def build_context(self, serialized_jobs):
+    """!Build context list for enqueue job
+    @param serialized_jobs List of DB jobs, serialized for Celery
+    """
+    context_list = []
+    kwargs = self.get_context_items()
+    for job in serialized_jobs:
+      context = {
+          'job': job,
+          'operation': self.operation,
+          'arch': self.dbt.session.arch,
+          'num_cu': self.dbt.session.num_cu,
+          'kwargs': kwargs,
+      }
+      context_list.append(context)
+
+    return context_list
 
   def celery_enqueue_call(self, context, q_name, task_id=False):
-    """Wrapper function for celery enqueue func"""
-    raise NotImplementedError('Not implemented')
+    """! Wrapper function for celery enqueue func
+    @param context serialized context for Celery job
+    @param q_name Name of custom Celery queue
+    @param task_id custom task ID for redis Key
+    """
+    Q_NAME = q_name  #pylint: disable=import-outside-toplevel,unused-variable,invalid-name,redefined-outer-name
```

Comment:
this is a side effect of needing the global Q_NAME at import time but not being able to assign it a value until runtime. Since imports are loaded before the program runs this workaround has to be done. Q_NAME here is a class variable and is visible where this assignment is made, so we dont need the global keyword here. We were OK with this design (although not ideal) because the other option would be to define it in an env variable and we did not want to add another env var for this.